app/helpers/reactjs_helper.rb

```diff
@@ -11,58 +10,60 @@ def react_component(name, props = {})
     content_tag('foreman-react-component', '', :name => name, :data => { props: props })
   end
 
-  def webpacked_plugins_with_global_css
-    global_css_tags(global_plugins_list).join.html_safe
+  def js_tags_for(requested_plugins)
+    requested_plugins.map do |plugin|
+      javascript_tag("window.tfm.tools.loadPluginModule('/webpack/#{plugin.to_s.tr('-', '_')}','#{plugin.to_s.tr('-', '_')}','./index');".html_safe)
+    end
   end
 
   def webpacked_plugins_js_for(*plugin_names)
     js_tags_for(select_requested_plugins(plugin_names)).join.html_safe
   end
 
-  def webpacked_plugins_with_global_js
-    global_js_tags(global_plugins_list).join.html_safe
+  def other_webpack_plugin(plugin_name, file)
+    javascript_tag("window.tfm.tools.loadPluginModule('/webpack/#{plugin_name.to_s.tr('-', '_')}','#{plugin_name.to_s.tr('-', '_')}','./#{file}_index');".html_safe)
   end
 
-  def webpacked_plugins_css_for(*plugin_names)
-    css_tags_for(select_requested_plugins(plugin_names)).join.html_safe
+  def global_js_tags(requested_plugins)
+    requested_plugins.map do |plugin|
+      plugin[:files].map do |file|
+        javascript_tag("window.tfm.tools.loadPluginModule('/webpack/#{plugin[:id].to_s.tr('-', '_')}','#{plugin[:id].to_s.tr('-', '_')}','./#{file}_index');".html_safe)
+      end
+    end
   end
 
-  def select_requested_plugins(plugin_names)
-    available_plugins = Foreman::Plugin.with_webpack.map(&:id)
-    missing_plugins = plugin_names - available_plugins
-    if missing_plugins.any?
-      logger.error { "Failed to include webpack assets for plugins: #{missing_plugins}" }
-      raise ::Foreman::Exception.new("Failed to include webpack assets for plugins: #{missing_plugins}") if Rails.env.development?
-    end
-    plugin_names & available_plugins
+  def webpacked_plugins_with_global_js
+    global_js_tags(global_plugins_list).join.html_safe
   end
 
-  def js_tags_for(requested_plugins)
-    requested_plugins.map do |plugin|
-      javascript_include_tag(*webpack_asset_paths(plugin.to_s, :extension => 'js'))
-    end
+  def webpacked_plugins_css_for(*plugin_names)
+    Foreman::Deprecation.deprecation_warning('3.9', '`webpacked_plugins_css_for` is deprecated, plugin css is already loaded.')
   end
 
-  def global_js_tags(requested_plugins)
-    requested_plugins.map do |plugin|
-      plugin[:files].map do |file|
-        javascript_include_tag(*webpack_asset_paths("#{plugin[:id]}:#{file}", :extension => 'js'))
-      end
+  def get_webpack_foreman_vendor_js
+    if ENV['RAILS_ENV'] == 'production'
+      javascript_include_tag("/webpack/#{File.basename(Dir.glob('public/webpack/foreman-vendor*production*js')[0])}")
+    else
+      javascript_include_tag("/webpack/#{File.basename(Dir.glob('public/webpack/foreman-vendor*development*js')[0])}")
```

Comment:
This implies in testing we also use the development version. Is that correct?

Reply:
The filename depends on `NODE_ENV` should I change `RAILS_ENV`  to `NODE_ENV`?
can `NODE_ENV` be something that is not `production` or `development`?

Reply:
As far as tests go we don't need webpack file to be loaded anyway (we don't do integration tests using `RAILS_ENV=test`). 

As I have mentioned before: since we need a single file there anyway, would it be OK to assert and prevent loading the app in case more than one file exists? Having the manifest file requires knowledge transfer between the webpack and rails parts of the code, and I try to simplify both webpack configuration and rails loading process.

In a long term, I think this part also should be part of the foreman federated module Maria already mentioned in the thread. This should remove the need to know the exact filename entirely.

Reply:
Shouldn't it be:
```suggestion
      javascript_include_tag("/webpack/#{File.basename(Dir.glob('public/webpack/foreman-vendor*#{ENV.fetch('NODE_ENV', 'production')}*js')[0])}")
```